src/napari_micromanager/_mda_handler.py

```diff
@@ -163,11 +194,15 @@ def _on_mda_frame(self, image: np.ndarray, event: MDAEvent) -> None:
             layer: Image = self.viewer.layers[layer_name]
             if not layer.visible:
                 layer.visible = True
-            # layer.reset_contrast_limits()
+            layer.reset_contrast_limits()
 
     def _on_mda_finished(self, sequence: MDASequence) -> None:
         # Save layer and add increment to save name.
+        self._mda_running = False
         if (meta := sequence.metadata.get(SEQUENCE_META_KEY)) is not None:
+            # TODO: make sure io thread finishes before saving
+            # thread workers don't seem to have .join method?
+            # self._io_t.join()
```

Comment:
@tlambert03 am I missing something here?

Reply:
no, `QRunnables` are slightly more restricted than true QThreads.  They don't have a `join()`... but they do have an `is_running()` method that you could wait on.  They also emit a `finished` signal that you could connect to a new method that does the saving.

```py
create_worker(self._watch_mda, _start_thread=True, _connect={'finished': self._on_io_finished})
```

Reply:
Is there an advantage to QRunnables that comes with the restrictions? I'm not sure I fully grasp the breakdown of thread vs runnable